# Primer ejercicio

## Enunciado

Un electrón con una capacidad de trabajo de $72090×10^{−19}$ J orbita de manera perpendicular a un campo magnético de __3250 G__. ¿Cuál es el radio de la órbita? ¿Y su frecuencia y periodo angular? Resuelve el mismo ejercicio para un antineutrón.

<img src="https://i.imgur.com/3cbe7if.gif" />

## Solución desarrollada 

Lo primero que vamos a desarrollar es el **concepto de Gauss**, se trata de una unidad que mide la fuerza que emite un campo magnético. Si esta es mayor, más fuerte será el campo y a su vez hay más espacio desde el exterior a la superficie de campo.

Ejemplos de uso de esta unidad de medida son, el campo magnético de la tierra es de 0,5 Gauss. Un imán pequeño tiene 100 Gauss y uno grande 15.000 Gauss

Otra cuestión a tener en cuenta es la **definición de joule**, a saber, la cantidad de trabajo que realiza una fuerza de un Newton en un metro de longitud situada en la misma dirección de dicha fuerza. Por tanto, la capacidad de trabajo define esa unidad de medida.

Además, debemos explicitar que **la masa de un electrón es 9.1 x 10-31 Kg**, se trata de una cantidad constante y que para nuestros cálculos también se da como valor conocido.

Finalmente explicar que cuando **una partícula cargada se mueve a una velocidad V** en un campo magnético normalmente se curva.

Con estas premisas paso a desarrollar el ejercicio planteado:





Inicializamos el entorno de Python para los cálculos

In [12]:
from sympy import*
import math
from sympy.physics.units import Quantity
Gs = Quantity("Gauss")
from sympy.physics.units.systems import SI
from sympy.physics.units import magnetic_flux_density
SI.set_quantity_dimension(Gs, magnetic_flux_density)
from sympy.physics.units import tesla
T= 10000*Gs
SI.set_quantity_scale_factor(Gs, 1E-4*tesla)
Gs=1E-4*tesla
B = 0.325*tesla
from sympy.physics.units import joule
J=symbols("juole")
E_c = 72090E-19 * joule
from sympy import solve, symbols, pi, Eq
r = symbols("radio")
v = symbols("velocidad")
from sympy.physics.units import electronvolt
from sympy.physics.units import volt
q = 1.602176565E-19
from sympy.physics.units import speed_of_light as c
from sympy.physics.units import kilogram
from sympy.physics.units import convert_to
from sympy.physics.units import meter
T = symbols("tesla")
import sympy.physics.units as sp_units
import scipy.constants as sp_constants
masa_electron  = sp_constants.electron_mass * sp_units.kg
carga_electron = sp_constants.electron_volt * sp_units.coulombs

print (" Mass (m):", masa_electron)
print (" Carga (q):", carga_electron)

 Mass (m): 9.1093837015e-31*kilogram
 Carga (q): 1.602176634e-19*coulomb


In [5]:
campo_B = (3250 * pintunits.gauss).to(pintunits.tesla).magnitude * sp_units.tesla
print "Campo (B): ", campo_B

SyntaxError: invalid syntax (<ipython-input-5-ebb032771eaf>, line 2)

**1. Convertimos los GAUSS enunciados en el ejercicio a Teslas.**

In [59]:
convert_to(3250*Gs, 1*tesla)

0.325*tesla

De ahí que en nuestro ejercicio el resultado es: 3250 G = **0,325 T**

2. Para realizar los cálculos **hemos de obtener la energía cinética** que se obtiene con la fórmula 

### $ E_c = \frac {1}{2}m v^2$

De ahí **despejamos la velocidad** ya que el resto de datos los hemos ido obteniendo del problema.

### $ 2E_C = m v^2 $    

y  tenemos:

### $ v = \frac {\sqrt{2 E_c(\frac{kg*m}{s^2}*m)}} {\sqrt{{m(kg)}} } $

In [34]:
v= 1.414213562 *sqrt(E_c)/sqrt(masa_electron) 
print ("v: ", v)

v:  125807928.837829*sqrt(joule)/sqrt(kilogram)


Etonce   $V=125807928.837829\frac{m}{s}$

3. Según la **ecuación de Lorentz** para realizar **el cálculo del radio debemos despejar** este parámetro obteniendo la siguiente formulación matemática sobre el planteamiento de origen.

## $ r = \frac {m v}{q B} $

Siendo **B la fuerza que ejerce el campo**, en el caso de nuestro ejercicio es ** 0,325 T** (hemos realizado la conversión con anterioridad).

In [39]:
r = masa_electron*v / q*B
print ("r:", r)

r: 0.000232591476646272*kilogram*tesla


In [ ]:
Entonces: r= 0.000232591476646272

4. Respecto a la **frecuencia angular** que es otro de los aspectos requeridos se obtiene con la fórmula: 

 ## $w = \frac {2 \pi} {t} $


## $t = \frac {2 \pi m}{q B}$

El tiempo necesario en la fórmula se consigue con la formulación matemática

## $v = \frac{s}{t}$

de la que se deduce que

## $ t = \frac {s}{v}$  



(siendo **s** el espacio recorrido).
Por lo tanto,

 ## $t = \frac{2\pi r}{v}$

In [40]:
t = (2*pi*r)/(v)
print ("t:",t)

t: 3.69565972647653e-12*pi*kilogram*tesla


Este cálculo nos aporta **el tiempo que tarda el electrón** en dar una vuelta.

Una vez encontrado el valor del tiempo se puede **calcular el resultado de la frecuencia angular** con la formula comentada al comienzo del epígrafe y conseguir el siguiente resultado


## $w = \frac {2 \pi} {t} $

In [41]:
w = (2*pi) /(t)
print("w:",w)

w: 541175364623.414/(kilogram*tesla)


## Calculando frecuencia y periodo angular

Si, además, queremos calcular la frecuencia y el periodo angular debemos saber que:
* Frecuencia (f) => Número de vueltas por segundo. Medido en Hertzios (Hz)
* Periodo    (T) => Tiempo que se tarda en dar una vuelta completa

Dado que el periodo es la inversa de la frecuencia:

## $T = \frac {1} {f} $

Y sabemos que el periodo angular es:

## $T = \frac {2 \pi r } {v} = \frac {2 \pi m} {q B} $


Por tanto, podemos calcular frecuencia y periodo angular:

In [13]:
T = (2 * sp_constants.pi * masa_electron) /(campo_B*carga_electron)
f = 1 / T;
print "Periodo (T): ", T
print "Frecuencia (f): ", f

Periodo (T):  1.09919587579355e-10*s
Frecuencia (f):  9097559607.18162/s


Si verificamos los cálculos utilizando la velocidad angular $\omega$:

Sabiendo que la velocidad velocidad angular es:

## $ \omega = \frac {q B} {m}$

La Frecuencia angular $f$:

##  $f = \frac {\omega} {2 \pi} $

Y el Periodo angular $T$:

## $T = \frac {2 \pi} {\omega} $


## Cálculos para un antineutrón

Un antineutrón es la antipartícula de un neutrón. Tiene la misma masa que éste y no tiene carga eléctrica, por tanto, no se verá afectado por un campo magnético.

In [44]:
# Masa de un neutron
masa_neutron  = sp_constants.neutron_mass * sp_units.kg
print ("Masa (m): ", masa_neutron)


Masa (m):  1.67492749804e-27*kilogram
